Classes/Controller/ImageRenderingController.php

```diff
@@ -78,16 +79,19 @@ public function renderImageAttributes(?string $content, array $conf = []): strin
                     $resourceFactory = GeneralUtility::makeInstance(ResourceFactory::class);
                     $systemImage     = $resourceFactory->getFileObject($fileUid);
 
-                    if ($imageSource !== $systemImage->getPublicUrl()) {
-                        // Source file is a processed image
+                    // check if there is a processed variant, if not, create one
+                    $processedFile = null;
+                    if ($systemImage instanceof File) {
```

Comment:
This check seems superfluous to me, `getFileObject()` always returns `File`, otherwise it would throw an exception.

https://api.typo3.org/11.5/class_t_y_p_o3_1_1_c_m_s_1_1_core_1_1_resource_1_1_resource_factory.html#aa11578c4934a73ea7239af6e5e788bc8

Reply:
yes, you are right, I will fix it

Reply:
done